perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ConsentsManagerBlImpl.java

```diff
@@ -343,7 +351,7 @@ private boolean hasValidConsent(PerunSession sess, Member member, ConsentHub con
 			return true;
 		} else if (revokedConsent != null && revokedConsent.getAttributes().containsAll(requiredAttributes)) {
 			return false;
-		} else if (unsignedConsent == null || !unsignedConsent.getAttributes().containsAll(requiredAttributes)) {
+		} else if (unsignedConsent == null || !unsignedConsent.getAttributes().containsAll(requiredAttributes) && consentEval) {
```

Comment:
Please add extra parentheses so the condition is more understandable.
```suggestion
		} else if ((unsignedConsent == null || !unsignedConsent.getAttributes().containsAll(requiredAttributes)) && consentEval) {
```